publication-commons/src/main/java/no/unit/nva/publication/service/impl/RestPublicationService.java

```diff
@@ -0,0 +1,166 @@
+    public static final String ERROR_MAPPING_PUBLICATION_TO_JSON = "Error mapping Publication to JSON";
+    public static final String IDENTIFIERS_NOT_EQUAL = "Identifier in request parameters '%s' "
+            + "is not equal to identifier in customer object '%s'";
+
+    private final ObjectMapper objectMapper;
+    private final HttpClient client;
+    private final String apiScheme;
+    private final String apiHost;
+
+    /**
+     * Constructor for RestPublicationService.
+     *
+     * @param objectMapper objectMapper
+     * @param client    client
+     * @param apiScheme apiScheme
+     * @param apiHost   apiHost
+     */
+    public RestPublicationService(HttpClient client, ObjectMapper objectMapper, String apiScheme, String apiHost) {
+        this.objectMapper = objectMapper;
+        this.client = client;
+        this.apiScheme = apiScheme;
+        this.apiHost = apiHost;
+    }
+
+    /**
+     * Constructor for RestPublicationService.
+     *
+     * @param client    client
+     * @param objectMapper objectMapper
+     * @param environment   environment
+     */
+    public RestPublicationService(HttpClient client, ObjectMapper objectMapper, Environment environment) {
+        this(client, objectMapper, environment.readEnv(API_SCHEME_ENV), environment.readEnv(API_HOST_ENV));
+    }
+
+    @Override
+    public Publication getPublication(UUID identifier, String authorization)
+            throws ApiGatewayException {
+        URI uri = UrlBuilder.empty()
+                .withScheme(apiScheme)
+                .withHost(apiHost)
+                .withPath(PATH + identifier.toString())
+                .toUri();
+
+        HttpRequest httpRequest = HttpRequest.newBuilder()
+                .uri(uri)
+                .header(ACCEPT, APPLICATION_JSON)
+                .header(AUTHORIZATION, authorization)
+                .GET()
+                .build();
+
+        try {
+            HttpResponse<String> httpResponse = client.send(httpRequest, HttpResponse.BodyHandlers.ofString());
+
+            JsonNode jsonNode = objectMapper.readTree(httpResponse.body());
+            JsonNode item0 = jsonNode.at(ITEMS_0);
+            if (item0.isMissingNode()) {
+                throw new NotFoundException("Publication not found for identifier: " + identifier);
+            }
+
+            return objectMapper.readValue(objectMapper.writeValueAsString(item0), Publication.class);
+        } catch (Exception e) {
+            throw new NoResponseException(ERROR_COMMUNICATING_WITH_REMOTE_SERVICE + uri.toString(), e);
+        }
+    }
+
+    @Override
+    public Publication updatePublication(UUID identifier, Publication publication, String authorization)
+            throws ApiGatewayException {
+        validateIdentifier(identifier, publication);
+        System.out.println("Sending request to modify resource " + identifier.toString());
```

Comment:
Use logger. Write now the logger we are using is not very different from System.out. But in the future we will use a proper logger that shows Class and Method information